From 18d43b79aa11e498d419c47c5c4eefd8908cce7c Mon Sep 17 00:00:00 2001
Message-ID: <18d43b79aa11e498d419c47c5c4eefd8908cce7c.1737500778.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1737500778.git.sharlatanus@gmail.com>
References: <cover.1737500778.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 21 Jan 2025 14:50:53 +0000
Subject: [PATCH 18/27] gnu: python-spectral-cube: Update to 0.6.6.

* gnu/packages/astronomy.scm (python-spectral-cube): Update to 0.6.6.
[source] <uri>: Adjust it as the name of PyPI archive has been changed.
[propagated-inputs]: Remove python-six; add python-aplpy,
python-packaging, python-pvextractor-bootstrap, and python-tqdm.
[native-inputs]: Remove python-pytest-astropy and python-pytest-xdist.

Change-Id: Idf63ee22a289cedd6f6ba7122a62ff93cf64571f
---
 gnu/packages/astronomy.scm | 27 +++++++++------------------
 1 file changed, 9 insertions(+), 18 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index aa9c9d2b111..1ad0bd4dc8c 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -4827,13 +4827,13 @@ (define-public python-sunpy-soar
 (define-public python-spectral-cube
   (package
     (name "python-spectral-cube")
-    (version "0.6.5")
+    (version "0.6.6")
     (source
      (origin
        (method url-fetch)
-       (uri (pypi-uri "spectral-cube" version))
+       (uri (pypi-uri "spectral_cube" version))
        (sha256
-        (base32 "0cqxgmk46gv2qa2kls8fph105lgwbwf13hvizh6w85mzgypyp740"))))
+        (base32 "0hz2pxc7fnxd1xr1n74ljjc84j25plnclp3y6jwg1banps360c3f"))))
     (build-system pyproject-build-system)
     (arguments
      (list
@@ -4842,33 +4842,24 @@ (define-public python-spectral-cube
       #:test-flags
       #~(list "--numprocesses" (number->string (parallel-job-count)))))
     (propagated-inputs
-     (list python-astropy
-           ;; XXX: Currently failing in upstream as it's optional silent
-           ;; until a new version is released, see:
-           ;; <https://github.com/aplpy/aplpy/issues/492>.
-           ;;
-           ;; python-aplpy
+     (list python-aplpy
+           python-astropy
            python-casa-formats-io
            python-dask
            python-distributed
            python-fsspec
-           ;; python-glue-core ; Not packed http://glueviz.org/, optional.
            python-joblib
            python-matplotlib
            python-numpy
+           python-packaging
+           python-pvextractor-bootstrap
            python-radio-beam
            python-reproject
            python-scipy
-           python-six
-           ;; python-yt ; Not packed https://yt-project.org/, optional.
+           python-tqdm
            python-zarr))
     (native-inputs
-     (list ;; XXX: Introduce cycle with pvextractor, listed as extra requiremnts
-           ;; in [noviz] option.
-           ;; python-pvextractor
-           python-pytest-astropy
-           python-pytest-xdist
-           python-regions
+     (list python-regions
            python-setuptools-scm
            python-wheel))
     (home-page "https://spectral-cube.readthedocs.io/en/latest/")
-- 
2.47.1

